src/GameLogic/PlayerActions/Skills/SummonPartySkillPlugin.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace MUnique.OpenMU.GameLogic.PlayerActions.Skills;
+
+using System.Runtime.InteropServices;
+using System.Threading;
+using MUnique.OpenMU.GameLogic.Views;
+using MUnique.OpenMU.GameLogic.Views.World;
+using MUnique.OpenMU.Pathfinding;
+using MUnique.OpenMU.PlugIns;
+
+/// <summary>
+/// The summon skill action for Dark Lord.
+/// </summary>
+[PlugIn(nameof(SummonPartySkillPlugin), "Handles the summon party skill of the dark lord class.")]
+[Guid("44e34497-c9e1-4c15-9388-589dfa3fa820")]
+public class SummonPartySkillPlugin : TargetedSkillPluginBase
+{
+    private static readonly int CountdownSeconds = 5;
+
+    /// <inheritdoc />
+    public override short Key => 63;
+
+    /// <inheritdoc />
+    public override async ValueTask PerformSkillAsync(Player player, IAttackable target, ushort skillId)
+    {
+        var skillEntry = player.SkillList?.GetSkill(skillId);
+        if (skillEntry?.Skill is null)
+        {
+            return;
+        }
+
+        if (player.Party is null)
+        {
+            return;
+        }
+
+        if (!await player.TryConsumeForSkillAsync(skillEntry.Skill).ConfigureAwait(false))
+        {
+            return;
+        }
+
+        await player.ForEachWorldObserverAsync<IShowSkillAnimationPlugIn>(p => p.ShowSkillAnimationAsync(player, player, skillEntry.Skill.Number, true), true).ConfigureAwait(false);
+
+        var cancellationTokenSource = new SkillCancellationTokenSource();
+        player.SkillCancelTokenSource = cancellationTokenSource;
+
+        _ = this.RunSummonPartyAsync(player, cancellationTokenSource);
+    }
+
+    private async ValueTask RunSummonPartyAsync(Player player, CancellationTokenSource cancellationTokenSource)
+    {
+        var cancellationToken = cancellationTokenSource.Token;
+        var targets = player.Party!.PartyList;
+        var targetPlayers = targets.OfType<Player>().Where(p => p != player);
+
+        try
+        {
+            for (var count = CountdownSeconds; count > 0; count--)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                foreach (var targetPlayer in targetPlayers)
+                {
+                    await targetPlayer.InvokeViewPlugInAsync<IChatViewPlugIn>(
+                        p => p.ChatMessageAsync($"Summoning in {count} second(s)...", player.Name, ChatMessageType.Party)).ConfigureAwait(false);
+                }
+
+                if (!player.IsAlive || player.IsAtSafezone())
+                {
+                    await player.Party.SendChatMessageAsync("Summoning canceled.", player.Name).ConfigureAwait(false);
+                    return;
+                }
+
+                await Task.Delay(1000, cancellationToken).ConfigureAwait(false);
+
+                targetPlayers = targetPlayers.Where(this.CanTargetBeTeleported);
```

Comment:
Sorry to bother you again 😅
The check `this.CanTargetBeTeleported` is not actually be executed inside of the loop, but when enumerating the players.
To fix that, `targetPlayers` should be a List, so just add a ToList() in the first line.
Then in the loop, call `targetPlayers.RemoveAll(t => !this.CanTargetBeTeleported(t))`

Reply:
I think it's good to check every countdown (check per one second) in the loop.  
This will update the targetPlayers that can be summoned.

Reply:
It's possible that for every second, a party member is excluded, then for the next second, another target is excluded, then after the timer expires, only the party members that are not violating any rules will be summoned.

Reply:
Filtering updated as suggested.